### PR TITLE
fix: avoid rw datarace on CompactionTask

### PR DESCRIPTION
**Pull request by SimFG**
- issue : #34523
 czs pr: #34718

### CODE REVIEW

**Comment by SimFG** [comment]
rerun ut

File: internal/datacoord/compaction.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ import (
 	"time"
 
 	"github.com/cockroachdb/errors"
+	"github.com/golang/protobuf/proto"
```
Comment (congqixia) [inline internal/datacoord/compaction.go:27]: please use `google.golang.org/protobuf/proto` instead
code check failed to detect this problem

**Review by congqixia — CHANGES_REQUESTED** [review]

File: internal/datacoord/compaction.go (hunk 1 above)
Comment (SimFG) [reply internal/datacoord/compaction.go]: has fixed it.

**Pull request closed without merging**